a3p-integration/proposals/z:acceptance/psm.test.js

```diff
@@ -0,0 +1,274 @@
+/* eslint-env node */
+/**
+ * @file The goal of this file is to implement a set of tests to make sure PSM works properly
+ *
+ * Here are the steps we want to take;
+ * 1 - Change swap fees and mint limit according to "psmTestSpecs" below
+ * 2 - Create a new user using agd.keys
+ * 3 - Fund new user with a stable coin from the VALIDATOR
+ *     - Do not provision manually
+ * 4 - Make sure new user is able to mint IST from PSM (fees are applied)
+ * 5 - Make sure new user can pay their debt and get their anchor (fees are applied)
+ * 6 - Make sure mint limit is adhered
+ */
+
+import test from 'ava';
+import {
+  getUser,
+  GOV1ADDR,
+  GOV2ADDR,
+  GOV3ADDR,
+  waitForBlock,
+} from '@agoric/synthetic-chain';
+import {
+  adjustBalancesIfNotProvisioned,
+  agopsPsm,
+  bankSend,
+  checkGovParams,
+  checkSwapExceedMintLimit,
+  checkSwapSucceeded,
+  checkUserInitializedSuccessfully,
+  getPsmMetrics,
+  implementPsmGovParamChange,
+  initializeNewUser,
+  maxMintBelowLimit,
+} from './test-lib/psm-lib.js';
+import { getBalances } from './test-lib/utils.js';
+
+// Export these from synthetic-chain?
+export const USDC_DENOM = process.env.USDC_DENOM
+  ? process.env.USDC_DENOM
+  : 'no-denom';
+export const PSM_PAIR = process.env.PSM_PAIR?.replace('.', '-');
```

Comment:
`.replace()` will fail if `PSM_PAIR` is undefined, right?

```suggestion
export const PSM_PAIR = NonNullish(process.env.PSM_PAIR).replace('.', '-');
```

Reply:
Resolved in https://github.com/Agoric/agoric-sdk/pull/10273/commits/11b336bb2c2fe58adc25a7a779ec6996d23e7d10